modules/pricer/src/main/java/com/opengamma/strata/pricer/sensitivity/CurveSensitivityUtils.java

```diff
@@ -10,23 +10,106 @@
 import java.time.LocalDate;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
 
 import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.currency.Currency;
 import com.opengamma.strata.collect.ArgChecker;
 import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
 import com.opengamma.strata.market.curve.CurveCurrencyParameterSensitivities;
 import com.opengamma.strata.market.curve.CurveCurrencyParameterSensitivity;
 import com.opengamma.strata.market.curve.CurveMetadata;
 import com.opengamma.strata.market.curve.CurveParameterMetadata;
+import com.opengamma.strata.market.curve.CurveParameterSize;
 import com.opengamma.strata.market.curve.DatedCurveParameterMetadata;
 import com.opengamma.strata.market.curve.DefaultCurveMetadata;
 import com.opengamma.strata.market.curve.meta.SimpleCurveNodeMetadata;
 import com.opengamma.strata.market.curve.meta.TenorCurveNodeMetadata;
+import com.opengamma.strata.math.impl.matrix.CommonsMatrixAlgebra;
+import com.opengamma.strata.math.impl.matrix.MatrixAlgebra;
+import com.opengamma.strata.product.ResolvedTrade;
 
 /**
  * Utilities to transform sensitivities.
  */
 public class CurveSensitivityUtils {
+  /**
+   * The matrix algebra used for matrix inversion.
+   */
+  private static final MatrixAlgebra MATRIX_ALGEBRA = new CommonsMatrixAlgebra();
+
+  /**
+   * Construct the inverse Jacobian matrix from the sensitivities of the trades market quotes to the curve parameters.
+   * <p>
+   * All the trades and sensitivities must be in the same currency. The data should be coherent with the
+   * market quote sensitivities passed in an order coherent with the list of curves.
+   * <p>
+   * For each trade describing the market quotes, the sensitivity provided should be the sensitivity of that
+   * market quote to the curve parameters.
+   * 
+   * @param order  the order in which the curves should be represented in the jacobian
+   * @param marketQuoteSensitivities  the market quotes sensitivity to the curve parameters
+   * @return inverse jacobian matrix, which correspond to the sensitivity of the parameters to the market quotes.
+   */
+  public static DoubleMatrix jacobianFromMarketQuoteSensitivities(
+      List<CurveParameterSize> order,
+      List<CurveCurrencyParameterSensitivities> marketQuoteSensitivities) {
```

Comment:
When parameters are over multiple lines, we have a blank line before the code starts.